docs/contributing.rst

```diff
@@ -0,0 +1,196 @@
+   `MDAnalysisData/adk_equilibrium.py`_, which becomes
+   :mod:`MDAnalysisData.adk_equilibrium`). In many cases you can copy an
+   existing module and adapt:
+   
+   - text: describe your dataset
+   - :data:`NAME`: name of the data set; will be used as a file name so do not use spaces etc
+   - :data:`DESCRIPTION`: filename of the description file (which contains
+     restructured text format, so needs to have suffix ``.rst``)
+   - :data:`ARCHIVE`: dictionary containing
+     :class:`~MDAnalysisData.base.RemoteFileMetadata` instances. Keys should
+     describe the file type. Typically
+
+     - *topology*: topology file (PSF, TPR, ...)
+     - *trajectory*: trajectory coordinate file (DCD, XTC, ...)
+     - *structure* (optional): system with single frame of coordinates
+       (typically PDB, GRO, CRD, ...)
+	  
+   - name of the :func:`fetch_{NAME}` function (where ``{NAME}`` is a suitable
+     name to access your dataset)
+   - docs of the :func:`fetch_{NAME}` function
+   - calculate and store the reference :ref:`SHA256 checksum <checksum>` as
+     described below
+     
+2. Add a description file (example:
+   `MDAnalysisData/descr/adk_equilibrium.rst`_); copy an existing file and
+   adapt. **Make sure to add license information.**
+3. Import your :func:`fetch_{NAME}` function in
+   `MDAnalysisData/datasets.py`_. ::
+
+      from .{MODULE_NAME} import fetch_{NAME}
+     
+4. Add documentation ``{NAME}.rst`` in restructured text format under `docs/`_
+   (take existing files as examples) and append ``{NAME}`` to the second
+   ``toctree`` section of the `docs/index.rst`_ file.
+
+   .. code-block:: reST
+
+      .. toctree::
+	 :maxdepth: 1
+	 :caption: Datasets
+	 :hidden:
+
+	 adk_equilibrium
+	 adk_transitions
+	 ...
+	 CG_fiber		   
+         {NAME}
+	 
+If your data set does not follow the same pattern as the example above (where
+each file is downloaded separately) then you have to write your own
+:func:`fetch_{NAME}` function. E.g., you might download a tar file and then
+unpack the file yourself. Use scikit-learn's `sklearn/datasets`_ as examples,
+make sure that your function sets appropriate attributes in the returned
+:class:`~MDAnalysisData.base.Bunch` of records, and fully document what is
+returned.
+
+
+.. _checksum:
+
+RemoteFileMetadata and SHA256 checksum
+======================================
+
+The :class:`~MDAnalysisData.base.RemoteFileMetadata` is used by
+:func:`~MDAnalysisData.base._fetch_remote` and it will check file integrity by
+computing a SHA256 checksum over each downloaded file with a stored reference
+checksum. **You must compute the reference checksum and store it in your**
+:class:`~MDAnalysisData.base.RemoteFileMetadata` data structure for each file.
+
+Typically you will have a local copy of the files during testing. You can
+compute the SHA256 for a file ``FILENAME`` with the following code::
+
+  python import MDAnalysisData.base
+  print(MDAnalysisData.base._sha256(FILENAME))
```

Comment:
Do we want to tell them to use sha256sum as an alternative or is that too platform specific?

Reply:
I'd keep it simple and just advertise one solution. If you know about sha256sum or any of the other tools that can do it, you'll do it anyway. This is at least a way that's guaranteed to work and be compatible with the code.

Reply:
Makes sense 👍🏽